kyo-data/shared/src/main/scala/kyo/Chunk.scala

```diff
@@ -354,9 +355,7 @@ sealed abstract class Chunk[A] extends Seq[A] derives CanEqual:
                 c.toArray
 
 end Chunk
```

Comment:
Have you also considered `Chunk` extending `StrictOptimizedSeqOps`?
It seems like the sibling of `StrictOptimizedSeqFactory`, but for the class itself, instead of the companion object.

For the record, ZIO's Chunk extends
 * `IndexedSeq`
 * `IndexedSeqOps`
 * `StrictOptimizedSeqOps`
 * `IterableFactoryDefaults`

https://github.com/zio/zio/blob/series/2.x/core/shared/src/main/scala-2.13%2B/zio/ChunkLike.scala#L40-L43

Reply:
We definitely should be extending `IndexedSeq`, not `Seq`. However, I will need to take a look at the implications of these other traits. From a quick look `StrictOptimizedSeqOps` simply uses the Builder to optimize the cost of each operator.

This is worth a followup to avoid further changes in this PR.

Reply:
Let's make sure we need those extensions. One of the main costs of scala collections is polymorphic interface dispatches. They require a linear search on the interfaces implemented by the object. More interfaces can incur into more overhead.